subprojects/core/src/main/java/org/gradle/api/internal/catalog/parser/TomlCatalogFileParser.java

```diff
@@ -129,7 +127,7 @@ private void parse() throws IOException {
             TomlTable pluginsTable = result.getTable(PLUGINS_KEY);
             Sets.SetView<String> unknownTle = Sets.difference(result.keySet(), TOP_LEVEL_ELEMENTS);
             if (!unknownTle.isEmpty()) {
-                throw throwVersionCatalogProblemException(problemServiceSupplier.get().create(builder ->
+                throw throwVersionCatalogProblemException((InternalProblems) problemServiceSupplier.get(), ((InternalProblems) problemServiceSupplier.get()).internalProblemReporter().create(builder ->
```

Comment:
❌ we should have a method for `(InternalProblems) problemServiceSupplier.get(), ((InternalProblems) problemServiceSupplier.get()).internalProblemReporter()` e.g. `getInternalReporter`